### PR TITLE
[NETBEANS-4857] prevent property change events during tree sync

### PR DESCRIPTION
**Pull request by MKuettner**
Prevent PropertyChangeEvents in ExplorerManager during tree synchronization if multiple OutlineViews share one single ExplorerManager.
See Jira Issue: https://issues.apache.org/jira/browse/NETBEANS-4857

### CODE REVIEW

File: platform/openide.explorer/src/org/openide/explorer/view/OutlineView.java [hunk 1]
```diff
@@ -876,6 +880,9 @@ final void synchronizeSelectedNodes(boolean scroll, Node... nodes) {
                 }
             }
         }
+        // [NETBEANS-4857]: re-activate property change events
+        outline.getSelectionModel().addListSelectionListener(managerListener);
```
Comment (eirikbakke) [inline platform/openide.explorer/src/org/openide/explorer/view/OutlineView.java:884]: But shouldn't there be at least one event processed in the end?

Might it be better to do outline.getSelectionModel().setValueIsAdjusting(true) at the beginning and setValueIsAdjusting(false) at the end? Does this work?

Comment (MKuettner) [reply platform/openide.explorer/src/org/openide/explorer/view/OutlineView.java:884]: The ExplorerManager.PROP_SELECTED_NODES property change event is already fired before _synchronizeSelectedNodes(..)_ method is called. So i think there is no need for any more property change event within the synchronize method.

But you are right setting _setValueIsAdjusting(true/false)_ is a much better solution. 
I just tested this and it works fine for me.

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]
Can I see a unitest? CCing @timboudreau

**Review by JaroslavTulach — APPROVED** [review]
A change with a test is always better:
* the test demonstrates the failure before the fix
* the test proves the fix "fixes something" (at least the test)
* the test prevents regressing to previous behavior in the future

You have my "go". Thanks for contributing to NetBeans.

**Review by eirikbakke — APPROVED** [review]
Thanks for all the tests!

**Comment by eirikbakke** [comment]
@MKuettner One technicality before we can merge this--could you ensure that the email address you want associated with this commit is visible on your Github profile (as a "public email" in the "profile" section of your settings)?

(When we merge the commits into the NetBeans codebase, they will become a single commit with author information taken from your github profile.)

**Comment by MKuettner** [comment]
@eirikbakke I changed my profile settings so my email address should be visible now.
Thanks for approving!

**Pull request merged**